docs/concepts/models/overview.md

```diff
@@ -354,6 +354,11 @@ Learn more about these properties and their default values in the [model configu
 ### gateway
 :   Specifies the gateway to use for the execution of this model. When not specified, the default gateway is used.
 
+### optimize
+:   Whether the model's SQL should be optimized. This attribute is `true` by default. Setting it
+to `false` causes SQLMesh to disable query canonicalization & simplification. This should be turned off only if the optimized query leads to errors such as surpassing text limit.
```

Comment:
@tobymao this feature is probably gonna be niche, but I wonder if we can optimize at load time to build the mapping schemas and then only switch off optimization when rendering a query to execute it. The motivation being that setting `optimize false` also impacts the column-level lineage functionality, so trying to see if there's a way to avoid it.

Reply:
i think that's ok, if they want lineage, they can fully qualify. i think the point of this flag is not not optimize at all, even at load time

Reply:
let's put a note saying here that doing this will disable column level lineage unless you manually qualify and expand all stars

Reply:
Yeah I can see what you mean, my hunch is also that this won't really be used _that_ much in practice, anyway. 

Though my understanding was that the issue is related to the payload's size when sending queries to the engine, so that wouldn't affect the loading stage.